### ISSUE TITLE
Send sign-ups that identify as having 11+ employees back to Classic. 

### ISSUE THREAD

**Issue by trjExpensify** [Expensify/App#53906]
### Background
- We’ve been working diligently for months on releasing the HybridApp. We’ve done that, so now switching between the NewDot experience and the OldDot experience in-app is seamless.
- We’ve started redirecting new sign-ups on web/mweb to NewDot if they identify as a company with <10 employees, as that’s our current focus.
- In NewDot, we have an onboarding modal that has a step for employee count.

### Problem
When all mobile sign-ups are put in the OldDot experience, it causes us to miss out on ~20% of new sign-up traffic, which prevents us from better understanding and optimising NewDot conversion on mobile.

### Solution
Let’s start putting all new native mobile sign-ups in the NewDot experience by default. From there, we’ll send any 11+ leads back to Classic. For consistency, we’ll do this for NewDot Web sign-ups as well. How that looks is as follows:

**HybridApp**
1. Download the Expensify app
2. Land in the NewDot experience (note: This requires the `hybridAppRedirect` beta, but the code exists to put new native mobile sign-ups in the newDot experience). 
3. Choose any 11+ option in the employee count step of the onboarding modal
4. User is switched to the Inbox page of the OldApp experience
5. `tryNewDot.classicRedirect.dismissed` is set to `true`.

**Web/mWeb**
1. Sign-up on [new.expensify.com](http://new.expensify.com/)
2. Choose any 11+ option in the employee count step of the onboarding modal
3. User is redirected to [expensify.com/inbox](http://expensify.com/inbox) in the same browser tab
4. `tryNewDot.classicRedirect.dismissed` is set to `true`.

**Comment by trjExpensify** [comment]
[Asking SWM](https://expensify.slack.com/archives/C07HPDRELLD/p1733913039418269?thread_ts=1733484433.968479&cid=C07HPDRELLD) about taking this one. It has a HybridApp element to it, and I suspect we will need a backend PR for updating the `tryNewDot.classicRedirect.dismissed` NVP to `true` on redirect when a 11+ option is selected.

**Comment by mateuuszzzzz** [comment]
Hi! I'll handle this issue 👌

**Comment by trjExpensify** [comment]
Dope, thanks!

**Comment by mateuuszzzzz** [comment]
I have a question regarding logic mentioned in the description. What about `Try New Expensify(OD)` and` Switch to Expensify Classic Buttons(ND)` buttons, should those buttons be visible all time for both cases <= 11 and 11+? 

Another question related to `hybridAppRedirect` beta. What is the state of this beta? Is it ready?

**Comment by trjExpensify** [comment]
Yeah, AFAIK it's ready. I have it on the @trj.chat domain if you want to test creating an account with any variation of tom+<insertSomething>@trj.chat. 

But yes, that beta should put you in the NewDot experience on HybridApp on sign-up and set `tryNewDot.classicRedirect.dismissed` to `false`. 

CC: @AndrewGable as well as he built it. :)

**Comment by trjExpensify** [comment]
Sounds good!

**Comment by mateuuszzzzz** [comment]
I left short demo and related question in [PR comment](https://github.com/Expensify/App/pull/54033#issuecomment-2538894635)

**Comment by trjExpensify** [comment]
Requested we fix the conflicts in [the PR](https://github.com/Expensify/App/pull/54033#issuecomment-2569020519), as we're in a position to merge this now.

**Comment by trjExpensify** [comment]
@mateuuszzzzz when testing the PR, I noticed it didn't have the test steps for web & mWeb from the OP of this issue:

<img width="850" alt="image" src="https://github.com/user-attachments/assets/3f6a4176-adf1-4215-ba59-932d85864401" />

The solution doesn't work on those platforms on staging either. Did you plan to create a separate PR for web?

**Comment by trjExpensify** [comment]
Quick bump [on this](https://github.com/Expensify/App/issues/53906#issuecomment-2578958107), @mateuuszzzzz!

**Comment by trjExpensify** [comment]
mobile PR merged, asked for a status update on Web from @mateuuszzzzz!

**Comment by mateuuszzzzz** [comment]
I linked draft of the PR

**Comment by trjExpensify** [comment]
No payments due here for the HybridApp PR, removing the labels.

**Comment by Julesssss** [comment]
Just leaving some notes after a chat with Tom.

The API is called despite it not doing anything in some cases. There are two flows: for users with no NVP set, we show the 'quick tip' page. For others we show the 'dismissed reason' UI.

- A new API would work, but ideally we add new logic to the `switchToNewDot` API that prevents NVP application to users who have not previous manually returned to classic with a reason
    - In theory, we would only set the NVP to true for users who don't have a dismissed reason
    - This logic is a bit flakey though, a second API might be preferred
- We will likely further change this logic in the future, so preparing to handle wider cases would be a good idea

**Comment by trjExpensify** [comment]
Small correction: 

>There are two flows: for users with no NVP set **OR classicRedirect.dismissed: true**, we show the 'quick tip' page. **For classicRedirect.dismissed: false** we show the 'dismissed reason' UI.

**Comment by Julesssss** [comment]
I will be picking this up today or Monday. [Here's](https://github.com/Expensify/App/issues/53906#issuecomment-2617086390) the latest plan.

**Comment by Julesssss** [comment]
Picking this up today (I'm mobile deployer this week which will kill some of my time though)

**Comment by Julesssss** [comment]
Hey @trjExpensify. I'm back on this and trying to setup my test cases. Can you please verify my understanding below for when the `RedirectToNewDot` API call will be used:

**Case A: Settings > Switch to Classic**

**1) User with no NVP or `dismissed: true`** 
- Shown 'Quick Tip'
- `SwitchToNewDot` call
- `tryNewDot` NVP is not modified in API
- Reason is NOT set in API

**2) User with `dismissed: false`** 
- Shown 'Switch to classic'
- `SwitchToNewDot` call
- `tryNewDot` NVP is updated to `dismissed: true` in API
- Reason is set in API

**Case B: New sign up**

**1) Switch to classic** 
-  User selects 11+ employees
- `SwitchToNewDot` call
- `tryNewDot` NVP is updated to `dismissed: true` in API
- Reason is NOT set in API

**2) Keep in NewDot**
-  User selects 10 or less employees
- `SwitchToNewDot` call
- `tryNewDot` NVP is updated to `dismissed: false` in API (we can probably keep existing value from when account was created)
- Reason is NOT set in API

_________

Assuming the above is correct, handling both case A & B in the API call is complex for little benefit and it's simpler to create distinct calls. I can still reuse functions in the API layer to reuse code where overlap occurs.

**Comment by trjExpensify** [comment]
>SwitchToNewDot

`SwitchToOldDot` right? 😅 Looking at the rest now.

**Comment by trjExpensify** [comment]
>Case B: new.expensify.com or HybridApp sign up

Suggested a change there, just to clarify the sign-up avenues where we would show the employee selector currently. 👍 

>2) Keep in NewDot
>....
> - tryNewDot NVP is updated to dismissed: false in API (we can probably keep existing value from when account was created)

Agreed, this isn't an update of the NVP per se, as it should be set to `dismissed: false` already. 

>Assuming the above is correct, handling both case A & B in the API call is complex for little benefit and it's simpler to create distinct calls. I can still reuse functions in the API layer to reuse code where overlap occurs.

Makes sense to me!

**Comment by Julesssss** [comment]
Great, thanks. New [PR](https://github.com/Expensify/Web-Expensify/pull/45472) is in progress. 

@mateuuszzzzz the API will be `SwitchToOldDotOnCompanySize` with one param for you to pass: [onboardinCompanySize](https://github.com/Expensify/App/blob/eaddd2eb9007c24e3c16a4ff349a2798e107cb67/src/CONST.ts#L260) as `companySize`, as we do [here](https://github.com/Expensify/App/blob/eaddd2eb9007c24e3c16a4ff349a2798e107cb67/src/pages/OnboardingWorkspaces/BaseOnboardingWorkspaces.tsx#L52).

**Comment by Julesssss** [comment]
> Thanks! So basically we should be able to use it for both web and hybrid app, right?

Yeah exactly. [The PR](https://github.com/Expensify/Web-Expensify/pull/45472) is in review and with the next deploy the API will be available on staging for your testing.

**Comment by Julesssss** [comment]
PR was merged, the API should be available on staging with Mondays deploy -- it doesn't look we'll have a deploy today.

**Comment by mateuuszzzzz** [comment]
Awesome!

**Comment by trjExpensify** [comment]
Niceee! Rory is our web deployer this week, so I expect this to go to staging on PST time today.

**Comment by trjExpensify** [comment]
@mateuuszzzzz the Web PR is on staging now!

**Comment by mateuuszzzzz** [comment]
Nice! I'll get to work on this right away

**Comment by trjExpensify** [comment]
Appreciated! We're pretty keen to get this deployed this week, so we can remove the hybridAppRedirect beta and push all new native sign-ups to NewDot.

**Comment by Julesssss** [comment]
I'm [struggling](https://github.com/Expensify/App/pull/55514) to verify this due to potentially another bug that prevents new account being routed to NewDot. Will share more infomation soon.

**Comment by trjExpensify** [comment]
Yeah bwoii, get on that beta or rinse a tom+<summin>@trj.chat account. 👍

**Comment by Julesssss** [comment]
Alright! PR merged. On the next deploy we can test on staging and prep for removing the beta.

**Comment by Julesssss** [comment]
@trjExpensify we're live on prod. Let's use this issue to remove the beta. Would you like to collect any additional test cases for the PR while I prepare it?
- HybridApp 11+ redirection
- HybridApp <10 redirection
- mWeb/Desktop 11+ redirection
- mWeb/Desktop <10 redirection

_Not due for payment._

**Comment by trjExpensify** [comment]
Those seem good to me!

**Comment by trjExpensify** [comment]
>Would you like to collect any additional test cases for the PR while I prepare it?

Can't we just `*` the `hybridAppRedirect` beta first in Betaify, and then remove it after a little while in case we need to roll back for some reason? I'm pretty sure that's the standard practice.

**Comment by Julesssss** [comment]
> I'm pretty sure that's the standard practice.
Right, yeah I'll do that now. Are we good to enable for all domains?

**Comment by trjExpensify** [comment]
Let's chat [in thread](https://expensify.slack.com/archives/C07HPDRELLD/p1740437183605109?thread_ts=1739830108.420329&cid=C07HPDRELLD).

**Comment by Julesssss** [comment]
We're on prod. @trjExpensify want to do the honours?

**Comment by trjExpensify** [comment]
thread for the breadcrumbs, running [into a crash](https://expensify.slack.com/archives/C07HPDRELLD/p1741129285649269?thread_ts=1741062946.604809&cid=C07HPDRELLD).

**Comment by Julesssss** [comment]
Crash resolved with build `9.1.9.1` -- the 11+ redirection is testing well for me for domains on the beta.

**Comment by trjExpensify** [comment]
Still crashing on that version for me. 👍

**Comment by Julesssss** [comment]
> Still crashing on that version for me. 👍

Resolved [here](https://github.com/Expensify/App/issues/57786), awaiting deployment to prod. Slack discussion [here](https://expensify.slack.com/archives/C07HPDRELLD/p1741139144920089?thread_ts=1741062946.604809&cid=C07HPDRELLD).

**Comment by trjExpensify** [comment]
Waiting on the release to get to the app store, it got rejected: https://expensify.slack.com/archives/C07J32337/p1741363345782309

**Comment by trjExpensify** [comment]
Apple accepted the build. Thread for updates this week is [here](https://expensify.slack.com/archives/C07HPDRELLD/p1741613573555869?thread_ts=1741571391.954329&cid=C07HPDRELLD), I've just put `+@gmail.com` on the `hybridAppRedirect` beta. If that goes okay, we'll move to `*` the beta in the next day or two.

**Comment by Julesssss** [comment]
HybridApp gmail sign-ups have been working nicely for me on both iOS and Android today.

**Comment by Julesssss** [comment]
Hey @mountiny. 

As you are mobile-deployer I just wanted to make you aware we have just enabled a beta which routes new `@gmail.com` HybridApp users to the NewDot experience. Please keep this issue in mind if you notice any issues on this weeks deploy blockers.  I'll be doing the same, thanks!

**Comment by mountiny** [comment]
ok, thanks! cc @cristipaval for visibility too

**Comment by Julesssss** [comment]
Everything seems good so far for new gmail users 🤞 We're on track to open up the beta tomorrow!

**Comment by Julesssss** [comment]
@trjExpensify I'm going to open it up to all, last call for a halt

**Comment by trjExpensify** [comment]
Shiiiiip it!

**Comment by Julesssss** [comment]
All new users will be redirected to the NewDot experience:

<img width="993" alt="Image" src="https://github.com/user-attachments/assets/9083c263-a4db-41f8-8f3e-7ef555639dd9" />

**Comment by trjExpensify** [comment]
Wahooo!

**Comment by trjExpensify** [comment]
Alright, I think we can close this. We'll want to add a regression test though, so I've created a [QA issue](https://github.com/Expensify/Expensify/issues/480291) for the below: 

**HybridApp**

1. Download the Expensify app
2. Land in the NewDot experience
3. Choose any 11+ option in the employee count step of the onboarding modal
4. **Verify** the user is switched to the Inbox page of the OldApp experience
5. Sign-in to expensify.com on the web with that same account
6. Open the JS console and paste the below
```
g_account.nameValuePairs.tryNewDot.classicRedirect.dismissed
```
7. **Verify** tryNewDot.classicRedirect.dismissed is set to `true`.


**Web/mWeb**
1. Sign-up on new.expensify.com
2. Choose "Manage my team's expenses"
3. Choose any 11+ option in the employee count step of the onboarding modal
4. **Verify** the user is redirected to expensify.com/inbox in the same browser tab
5. Open the JS console and paste the below 
```
g_account.nameValuePairs.tryNewDot.classicRedirect.dismissed
```
6. **Verify** `tryNewDot.classicRedirect.dismissed` is set to `true`.

**Comment by Julesssss** [comment]
> We'll want to add a regression test though, so I've created a https://github.com/Expensify/Expensify/issues/480291 for the below:

Nice, thanks

**Comment by trjExpensify** [comment]
Payment summary as follows: 

- $250 to @rushatgabhane for the C+ review

**Comment by garrettmknight** [comment]
$250 approved for @rushatgabhane
